matching-core/src/main/java/com/graphhopper/matching/MapMatching.java

```diff
@@ -166,6 +166,8 @@ public MatchResult doWork(List<GPXEntry> gpxList) {
         // Compute all candidates first.
         // TODO: Generate candidates on-the-fly within computeViterbiSequence() if this does not
         // degrade performance.
+        // @kodonnell: it's not currently possible as that'd mean calling queryGraph.lookup
```

Comment:
I think that looking up all candidates at once might be a problem for map matching very long GPS traces, e.g. from Munich to Hamburg. But maybe it's better to create an issue for this than having this TODO in the code.